packages/jest-runtime/src/index.ts

```diff
@@ -1454,8 +1464,10 @@ export default class Runtime {
         return moduleRegistry.get(key) || null;
       },
     });
-
-    module.paths = this._resolver.getModulePaths(module.path);
+    module.paths = [
```

Comment:
While I haven't profiled this and can't know the perf impact for sure, this is a code path that runs very often, so I'd rather be safe and do something like
```js
    module.paths = this._resolver.getModulePaths(module.path);
    if(moduleName) module.paths=[...module.paths, ...this._resolver.getGlobalPaths(etc)]
```
But read the other comment first, perhaps this code might change anyway

Reply:
Also, cc @SimenB who knows more about the ESM impl cause I don't know all implications changing `module.paths` might have

Reply:
`module.*` is unused in ESM